bson/bsonrw/value_writer.go

```diff
@@ -47,11 +47,9 @@ func NewBSONValueWriterPool() *BSONValueWriterPool {
 // Get retrieves a BSON ValueWriter from the pool and resets it to use w as the destination.
 func (bvwp *BSONValueWriterPool) Get(w io.Writer) ValueWriter {
 	vw := bvwp.pool.Get().(*valueWriter)
-	if writer, ok := w.(*SliceWriter); ok {
-		vw.reset(*writer)
-		vw.w = writer
-		return vw
-	}
+
+	// TODO: Having to call reset here with the same buffer doesn't really make sense.
```

Comment:
I think it could be worth including as part of an existing ticket, but it's not worth refactoring on its own. Some other methods also call `reset(buf)` with a distinct buffer, so the refactor seems like it could be substantial for a small clarity benefit. I'll check if there's an existing ticket that makes sense to tag.

Reply:
I wasn't able to find a ticket that would have this change in-scope. I'll leave it here as commentary for the next person who updates this code.